service/history/events/cache.go

```diff
@@ -166,6 +186,7 @@ func (e *CacheImpl) DeleteEvent(namespaceID, workflowID, runID string, eventID i
 	sw := e.metricsClient.StartTimer(metrics.EventsCacheDeleteEventScope, metrics.CacheLatency)
 	defer sw.Stop()
 
+	e.validateIds(namespaceID, workflowID, runID, eventID)
```

Comment:
didn't check result?

Reply:
In this case we're just using it for the side effect (logging a warning). Deleting an invalid key from the cache can only be a good thing, so I allowed it.

Also, note that there are no calls to `DeleteEvent` in the codebase